compiler/lib/effects.ml

```diff
@@ -1140,6 +1140,7 @@ let f ~flow_info ~live_vars p =
   let p = split_blocks ~cps_needed p in
   let p, trampolined_calls, in_cps = cps_transform ~live_vars ~flow_info ~cps_needed p in
   if Debug.find "times" () then Format.eprintf "  effects: %a@." Timer.print t;
+  let p = Deadcode.remove_unused_blocks p in
```

Comment:
@OlivierNicole , @vouillon , is it expected that the effects pass produces unused  blocks ?

Reply:
I would expect so, yes. It creates a bunch of new blocks when creating all the continuation closures.

Reply:
@hhugo How do you get it to produce unused blocks? I removed this line and `dune build @runtest-js --profile with-effects` still runs without error.

Reply:
OCAMLRUNPARAM=b=1 dune exec js_of_ocaml `which ocamlc.byte` --  -o ocamlc.js --opt 1  --debug stats  --debug invariant --effects double-translation